### PR TITLE
Upgrade tar dependency to v7.x

### PR DESCRIPTION
**Pull request by WtfJoke**
Dropping support for node 6 and 8 (as tar v6 drops that support as well).
Addresses [CVE-2024-28863](https://github.com/advisories/GHSA-f5x3-32g6-xq36)

I did not notice any breaking changes in the api of tar (as it still works when tested it using our project).
The only [breaking change library side is](https://github.com/isaacs/node-tar/blob/main/CHANGELOG.md) that you can not use it in node 6 and 8 anymore (but I dont think anybody is on that version anymore).

I've verified it works by running following commands:
1. `node examples/simple.js` (I've added a simple setTimeout function which waits 2min before shutting it down, see [here](https://github.com/WtfJoke/dynamodb-local/commit/dfc92118a3c0f9dc7e96a07adc1b90d0790d19dc))
2. `aws dynamodb list-tables --endpoint-url http://localhost:8000`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "debug": "~4.3.1",
     "mkdirp": "~0.5.0",
     "q": "~1.4.1",
-    "tar": "^4.4.19"
+    "tar": "^7.4.3"
```
Comment (rynop) [inline package.json]: I'm not a fan of `^` as it can introduce breaking changes
```suggestion
    "tar": "~7.4.3"
```

**Review by rynop — CHANGES_REQUESTED** [review]
LGTM after comment!

**Comment by WtfJoke** [comment]
Thanks for your fast answer. Commited your suggested change with your comment! Looking forward to merge and release!

**Review by rynop — APPROVED** [review]

**Comment by rynop** [comment]
Version `0.0.35` published: https://www.npmjs.com/package/dynamodb-local

thanks!

**Comment by WtfJoke** [comment]
Thank you for the fast release <3

**Pull request merged**